assets/js/frontend/course-video/video-blocks-manager.js

```diff
@@ -0,0 +1,154 @@
+/**
+ * The Course Video Progression settings.
+ */
+const {
+	courseVideoRequired,
+	courseVideoAutoComplete,
+	courseVideoAutoPause,
+} = window.sensei.courseVideoSettings;
+
+/**
+ * Map of videos.
+ */
+const videos = {};
+
+/**
+ * Registers the videos to manage.
+ * @param {Object}   video
+ * @param {Function} video.pauseVideo              The function that pauses the video when invoked.
+ * @param {Function} video.registerVideoEndHandler Accepts a callback that is invoked when the video ends.
+ * @param {string}   video.url                     The source url of the video. Used as an id.
+ */
+export const registerVideo = ( {
+	pauseVideo = () => {},
+	registerVideoEndHandler = () => {},
+	url = '',
+} ) => {
+	const blocksStore = window.sensei?.store?.blocks;
+	if ( courseVideoRequired ) {
+		videos[ url ] = { pauseVideo, completed: false };
+		if ( blocksStore ) {
+			blocksStore.setAttributes( url, {
```

Comment:
I am thinking that having Sensei core rely on Pro's code might be a bit dangerous. Did you consider using a filter? We could for example pass the functions in `video-blocks-manager.js` in a filter and then in Pro have some logic which checks if the store exists and if it doesn't fallback to the original functions.

Reply:
That's actually a very good idea! Looks like gutenberg publishes the hooks api in a separate package. We can expose hooks from the Sensei Pro side and make Sensei core simply attach to those if available.

Will update! 👍